lib/kcerts/ssh.go

```diff
@@ -99,6 +99,31 @@ func (a SSHAgent) Valid() bool {
 	return err == nil
 }
 
+// Principals returns a map where the keys are the CA's PKS and the certs identities are the values
+func (a SSHAgent) Principals() (map[string][]string, error) {
+	conn, err := net.Dial("unix", a.Socket)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	keys, err := agent.NewClient(conn).List()
+	if err != nil {
+		return nil, err
+	}
+	toReturn := map[string][]string{}
+	for _, key := range keys {
+		p, err := ssh.ParsePublicKey(key.Marshal())
+		if err != nil {
+			return nil, err
```

Comment:
minor nit: the agent could be an arbitrary ssh-agent from 1972, or 2045, when our code will still be running. Rather than return error and fail if the agent handed us something we cannot parse, I would recommend continuing. Use mutlierror to return an error? Clarify that toReturn can contain some values even in case of error? Or return 3 values: valid keys, invalid keys, err

Reply:
So I toyed with the idea of ignoring errors here, but I decided that's why I silently ignore public keys that are not certificates. key.Marshall() is "supposed" to return the Public key always, so if it doesn't then there's something funky going on. 

Totally fine to ignore it too since there's no harm in it, but those are my thoughts. I'll put in a continue instead

Reply:
+1! You can just say in the doc of the function that Principals returns only certificates that enkit auth recognizes.
Adding a comment there.